File: 11-container-with-most-water/container-with-most-water.js
/**
 * @param {number[]} height
 * @return {number}
 */
var maxArea = function(height) {
    let n = height.length;
    let left=0;
    let right = n-1;
    let max =0;
    while(left<right){
        let minHeight = Math.min(height[left], height[right]);
        let currentArea = minHeight * (right-left);
        max = Math.max(max, currentArea);
        if(height[left]<height[right]){
            left++;
        } else{
            right--;
        }
    }
    return max;
};

File: 11-container-with-most-water/container-with-most-water.java
class Solution {
    public int maxArea(int[] height) {
        int left = 0;
        int right = height.length-1;
        int result=0;
        while(left<right){
            if(height[left]<height[right]){
                int currResult = height[left]*(right-left);
                if(currResult>result) result = currResult;
                left++;
            } else{
                int currResult = height[right]*(right-left);
                if(currResult>result) result = currResult;
                right--;   
            }
        }
        return result;
    }
}